admin/cluster_cleanup.py

```diff
@@ -0,0 +1,76 @@
+
+import sys
+
+from itertools import repeat
+
+from twisted.internet.defer import inlineCallbacks
+from twisted.python.filepath import FilePath
+from twisted.python.usage import Options, UsageError
+
+from flocker.apiclient import FlockerClient
+from flocker.common import gather_deferreds, loop_until
+from flocker.control.httpapi import REST_API_PORT
+
+
+class ScriptOptions(Options):
+    optParameters = [
+        ['control-node', None, None,
+         "The address of the cluster's control node"],
+        ['cert-directory', None, None,
+         "The directory containing the cluster certificates"],
+    ]
+
+    def postOptions(self):
+        if not self['control-node']:
+            raise UsageError("Control node address must be provided.")
+        if not self['cert-directory']:
+            raise UsageError("Certificates directory must be provided.")
+
+
+@inlineCallbacks
+def main(reactor, args, base_path, top_level):
+    try:
+        options = ScriptOptions()
+        options.parseOptions(args)
+    except UsageError as e:
+        sys.stderr.write(e.args[0])
+        sys.stderr.write('\n\n')
+        sys.stderr.write(options.getSynopsis())
+        sys.stderr.write('\n')
+        sys.stderr.write(options.getUsage())
+        raise SystemExit(1)
+
+    certificates_path = FilePath(options['cert-directory'])
+    cluster_cert = certificates_path.child(b"cluster.crt")
+    user_cert = certificates_path.child(b"user.crt")
+    user_key = certificates_path.child(b"user.key")
+    client = FlockerClient(reactor, options['control-node'], REST_API_PORT,
+                           cluster_cert, user_cert, user_key)
+
+    containers = yield client.list_containers_configuration()
+    results = []
+    for container in containers:
+        print "deleting container", container.name
+        results.append(client.delete_container(container.name))
+    yield gather_deferreds(results)
+
+    def containers_deleted():
+        d = client.list_containers_configuration()
+        d.addCallback(lambda containers: not containers)
+        return d
+
+    yield loop_until(client._reactor, containers_deleted, repeat(1, 300))
```

Comment:
Any numbers that are fixed in the source code, but could sensibly change to another value, should be constants.